dialogy/plugins/text/intent_entity_mutator/__init__.py

```diff
@@ -0,0 +1,309 @@
+
+        rules_ = rules[const.BASE_KEY]
+
+        permissible_transcript_functional_condition_keys = list(
+            transcript_functions_map.keys()
+        )
+
+        for rule in rules_:
+            rule_keys = sorted(list(rule.keys()))
+            if rule_keys != const.MANDATORY_RULE_KEYS:
+                raise ValueError(
+                    f"The rule is either missing some keyword or incorrect keyword entered. \
+                        Received: {rule_keys}. Expected: {const.MANDATORY_RULE_KEYS} \
+                            {rule}"
+                )
+
+            mutate = rule[const.MUTATE]
+            if mutate not in [const.INTENT, const.ENTITY]:
+                raise ValueError(
+                    f"The mutate keyword received incorrect value. \
+                        Received {mutate}. Expected: intent or entity \
+                            {rule}"
+                )
+
+            conditions = rule[const.CONDITIONS]
+
+            for primitive, search_lists in conditions.items():
+                if primitive not in const.BASE_CONDITION_PRIMITIVES:
+                    raise ValueError(
+                        f"Received invalid primitive in the rules. Received: {primitive}.  \
+                            Expected from: {const.BASE_ENTITY_PRIMITIVES} \
+                                {rule}"
+                    )
+
+                if search_lists:
+
+                    if primitive == const.ENTITY:
+                        received_entity_primitives = list(
+                            search_lists.keys()
+                        )  # Get keys such as [dim, type, ..]
+                        for ent_primitives in received_entity_primitives:
+                            if ent_primitives not in const.BASE_ENTITY_PRIMITIVES:
+                                raise ValueError(
+                                    f"The rule did not receive correct entity sub_type. Received: {ent_primitives} \
+                                        Expected: {const.BASE_ENTITY_PRIMITIVES} \
+                                            {rule}"
+                                )
+
+                    else:
+                        received_primitive_keys = list(
+                            search_lists.keys()
+                        )  # list of dictionaries. [{in: [], not_in: [], }]
+                        for key in received_primitive_keys:
+                            if isinstance(search_lists[key], list):
+                                if key not in const.SUB_PRIMITIVES:
+                                    raise ValueError(
+                                        f"The rule did not receive the correct list type. Received: {key} \
+                                            Expected: {const.SUB_PRIMITIVES} \
+                                                {rule}"
+                                    )
+
+                            else:  # boolean instance -> Transcript cases
```

Comment:
I think this `else` needs one more case where a functional condition is applied on non-transcript primitive. The below `if` assumes that the primitive is `transcript` but that might not be true when the control enters this else block.